src/gg.regression.unity.bots/Editor/Scripts/ActionManager/RGActionAnalysis.cs

```diff
@@ -562,20 +623,71 @@ public override void VisitInvocationExpression(InvocationExpressionSyntax node)
                             var btnArg = node.ArgumentList.Arguments[0];
                             foreach (var btnNameFunc in FindCandidateLiteralFuncs<string>(btnArg.Expression))
                             {
-                                string[] path = { objectType.FullName, $"Input.{methodName}({btnNameFunc})" };
-                                AddAction(new LegacyButtonAction(path, objectType, btnNameFunc));
+                                string[] path = GetActionPathFromSyntaxNode(objectType, node);
+                                AddAction(new LegacyButtonAction(path, objectType, btnNameFunc), node);
                             }
                             break;
                         }
                     }
+                } else if (containingType == typeof(Physics) || containingType == typeof(Physics2D))
+                {
+                    MousePositionType posType = containingType == typeof(Physics)
+                        ? MousePositionType.COLLIDER_3D
+                        : MousePositionType.COLLIDER_2D;
+                    
+                    string methodName = methodSymbol.Name;
+                    // Raycast
+                    switch (methodName)
+                    {
+                        case "Raycast":
+                        case "RaycastAll":
+                        case "RaycastNonAlloc":
```

Comment:
Additional high level comments on what is happening in these code blocks will be useful for future reviewers.

Reply:
Added more comments to the RGActionAnalysis class overall (as well as these code blocks as well).